--- origsrc/evolution-data-server-2.24.2/configure.in	2008-11-12 00:27:34.000000000 -0600
+++ src/evolution-data-server-2.24.2/configure.in	2008-12-14 02:14:42.759515700 -0600
@@ -260,8 +276,8 @@
   dynamic_libdb=yes, dynamic_libdb=no)
 
 if test "x${dynamic_libdb}" = "xyes"; then
-        DB_CFLAGS="-I$withval/include"
-        DB_LIBS="-L$withval/lib -ldb"
+        DB_CFLAGS="-I$withval/include/db4.1"
+        DB_LIBS="-L$withval/lib -ldb-4.1"
 
         AC_MSG_CHECKING([Berkeley DB])
         save_cflags=$CFLAGS; CFLAGS=$DB_CFLAGS
