--- origsrc/evolution-data-server-2.28.1/camel/providers/sendmail/Makefile.am	2009-10-17 18:08:43.000000000 -0500
+++ src/evolution-data-server-2.28.1/camel/providers/sendmail/Makefile.am	2009-10-29 17:12:10.989888700 -0500
@@ -20,7 +20,13 @@ libcamelsendmail_la_SOURCES = 			\
 noinst_HEADERS =		\
 	camel-sendmail-transport.h
 
-libcamelsendmail_la_LDFLAGS = -avoid-version -module
+libcamelsendmail_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
+
+libcamelsendmail_la_LIBADD = \
+	$(top_builddir)/libedataserver/libedataserver-${API_VERSION}.la \
+	$(top_builddir)/camel/libcamel-provider-1.2.la          \
+	$(top_builddir)/camel/libcamel-1.2.la               \
+	$(CAMEL_LIBS)
 
 libcamelsendmail_la_LIBADD = \
 	$(top_builddir)/camel/libcamel-1.2.la				\
--- origsrc/evolution-data-server-2.28.1/configure.ac	2009-10-17 20:48:03.000000000 -0500
+++ src/evolution-data-server-2.28.1/configure.ac	2009-10-29 17:14:43.630099400 -0500
@@ -229,30 +229,45 @@ PKG_PROG_PKG_CONFIG
 dnl Check for Win32
 AC_MSG_CHECKING([for Win32])
 case "$host" in
+*-cygwin*)
+	platform_win32=yes
+	os_win32=no
+	NO_UNDEFINED='-no-undefined'
+	DL_LIB=''
+	SOFTOKN3_LIB=''
+	LIBEXECDIR_IN_SERVER_FILE="$libexecdir"
+	SOPREFIX='cyg'
+	;;
 *-mingw*)
+	platform_win32=yes
 	os_win32=yes
 	NO_UNDEFINED='-no-undefined'
 	SOCKET_LIBS='-lws2_32 -ldnsapi'
 	DL_LIB=''
 	SOFTOKN3_LIB=''
 	LIBEXECDIR_IN_SERVER_FILE='../../../libexec'
+	SOPREFIX='lib'
 	AC_CACHE_VAL(ac_cv_have_addrinfo, [ac_cv_have_addrinfo=yes])
 	AC_DEFINE(_WIN32_WINNT, 0x501, [To get getaddrinfo etc declarations])
 	;;
 *)
+	platform_win32=no
 	os_win32=no
 	NO_UNDEFINED=''
 	SOCKET_LIBS=''
 	DL_LIB='-ldl'
 	SOFTOKN3_LIB='-lsoftokn3'
 	LIBEXECDIR_IN_SERVER_FILE="$libexecdir"
+	SOPREFIX='lib'
 	;;
 esac
 AC_MSG_RESULT([$os_win32])
+AM_CONDITIONAL(PLATFORM_WIN32, [test $platform_win32 = yes])
 AM_CONDITIONAL(OS_WIN32, [test $os_win32 = yes])
 AC_SUBST(NO_UNDEFINED)
 AC_SUBST(SOCKET_LIBS)
 AC_SUBST(LIBEXECDIR_IN_SERVER_FILE)
+AC_SUBST(SOPREFIX)
 
 dnl Check for posix compatible sys/wait.h
 AC_HEADER_SYS_WAIT
--- origsrc/evolution-data-server-2.28.1/servers/exchange/storage/Makefile.am	2009-10-17 18:08:43.000000000 -0500
+++ src/evolution-data-server-2.28.1/servers/exchange/storage/Makefile.am	2009-10-29 17:12:14.089893000 -0500
@@ -1,6 +1,6 @@
 # Fix the code to not use E_DATA_SERVER_UI_CFLAGS
 
-if OS_WIN32
+if PLATFORM_WIN32
 WIN32_BOOTSTRAP_LIBS =					\
 	$(top_builddir)/win32/libedataserverui-1.2.la
 endif
--- origsrc/evolution-data-server-2.28.1/win32/Makefile.am	2009-08-22 14:25:17.000000000 -0500
+++ src/evolution-data-server-2.28.1/win32/Makefile.am	2009-10-29 17:12:14.099893000 -0500
@@ -2,7 +2,7 @@ EXTRA_DIST =				\
 	dummy.la			\
 	libedataserverui.def
 
-if OS_WIN32
+if PLATFORM_WIN32
 BOOTSTRAP_LIBS =			\
 	libedataserverui-1.2.la
 endif
@@ -12,8 +12,8 @@ noinst_DATA = $(BOOTSTRAP_LIBS)
 libedataserverui_current_minus_age = `expr $(LIBEDATASERVERUI_CURRENT) - $(LIBEDATASERVERUI_AGE)`
 
 libedataserverui-1.2.la: dummy.la libedataserverui.def
-	sed -e s!%DLL%!libedataserverui-1.2-$(libedataserverui_current_minus_age)! -e s!%LIB%!libedataserverui-1.2! -e s!%PFX%!$(prefix)! <dummy.la >$@
+	sed -e s!%DLL%!$(SOPREFIX)edataserverui-1.2-$(libedataserverui_current_minus_age)! -e s!%LIB%!libedataserverui-1.2! -e s!%PFX%!$(prefix)! < $(srcdir)/dummy.la >$@
 	mkdir -p .libs
-	$(DLLTOOL) --output-lib=.libs/libedataserverui-1.2.dll.a --dllname=libedataserverui-1.2-$(libedataserverui_current_minus_age).dll --input-def=libedataserverui.def
+	$(DLLTOOL) --output-lib=.libs/libedataserverui-1.2.dll.a --dllname=$(SOPREFIX)edataserverui-1.2-$(libedataserverui_current_minus_age).dll --input-def=$(srcdir)/libedataserverui.def
 
 CLEANFILES = $(BOOTSTRAP_LIBS)
