--- origsrc/evolution-data-server-2.30.1/camel/providers/sendmail/Makefile.am	2010-03-18 09:14:32.000000000 -0500
+++ src/evolution-data-server-2.30.1/camel/providers/sendmail/Makefile.am	2010-06-07 08:04:47.558501500 -0500
@@ -19,7 +19,7 @@ libcamelsendmail_la_SOURCES = 			\
 noinst_HEADERS =		\
 	camel-sendmail-transport.h
 
-libcamelsendmail_la_LDFLAGS = -avoid-version -module \
+libcamelsendmail_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED) \
 	$(CODE_COVERAGE_LDFLAGS) \
 	$(NULL)

--- origsrc/evolution-data-server-2.30.1/configure.ac	2010-04-05 19:07:50.000000000 -0500
+++ src/evolution-data-server-2.30.1/configure.ac	2010-06-07 08:20:43.048152400 -0500
@@ -247,6 +247,13 @@ dnl Check for Win32
 dnl ******************************
 AC_MSG_CHECKING([for Win32])
 case "$host" in
+*-cygwin*)
+	os_win32='no'
+	NO_UNDEFINED='-no-undefined'
+	SOCKET_LIBS=''
+	DL_LIB=''
+	LIBEXECDIR_IN_SERVER_FILE="$libexecdir"
+	;;
 *-mingw*)
 	os_win32='yes'
 	NO_UNDEFINED='-no-undefined'
--- origsrc/evolution-data-server-3.6.2/libebackend/Makefile.am	2012-09-16 10:05:39.000000000 -0500
+++ src/evolution-data-server-3.6.2/libebackend/Makefile.am	2012-12-25 23:04:35.247902900 -0600
@@ -56,7 +56,6 @@ libebackend_1_2_la_SOURCES =		\
 
 libebackend_1_2_la_LIBADD = 				\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la \
-	$(top_builddir)/private/libedbus-private.la	\
 	$(E_BACKEND_LIBS)				\
 	$(CAMEL_LIBS)					\
 	$(SQLITE3_LIBS)					\
