--- origsrc/evolution-data-server-3.10.4/addressbook/backends/file/e-book-backend-file.c	2013-12-08 12:43:17.000000000 -0600
+++ src/evolution-data-server-3.10.4/addressbook/backends/file/e-book-backend-file.c	2014-07-14 18:28:05.777056700 -0500
@@ -1575,9 +1575,9 @@ book_backend_file_get_direct_book (EBook
 
 	/* Support in-tree testing / relocated modules */
 	if (modules_env)
-		backend_path = g_build_filename (modules_env, "libebookbackendfile.so", NULL);
+		backend_path = g_module_build_path (modules_env, "ebookbackendfile");
 	else
-		backend_path = g_build_filename (BACKENDDIR, "libebookbackendfile.so", NULL);
+		backend_path = g_module_build_path (BACKENDDIR, "ebookbackendfile");
 	direct = e_data_book_direct_new (backend_path, "EBookBackendFileFactory", dirname);
 
 	g_free (backend_path);
