--- origsrc/evolution-data-server-3.20.6/configure.ac	2017-03-06 21:30:45.085986500 -0600
+++ src/evolution-data-server-3.20.6/configure.ac	2017-03-06 21:38:32.034854200 -0600
@@ -58,7 +58,7 @@ m4_define([goa_minimum_version], [3.8])
 m4_define([gweather_minimum_version], [3.10])
 m4_define([libaccounts_glib_minimum_version], [1.4])
 m4_define([libsignon_glib_minimum_version], [1.8])
-m4_define([webkitgtk_minimum_version], [2.4.9])
+m4_define([webkitgtk_minimum_version], [2.0.4])
 m4_define([json_glib_minimum_version], [1.0.4])
 
 AC_SUBST([BASE_VERSION],[base_version])
