--- origsrc/evolution-data-server-2.22.2/calendar/backends/caldav/Makefile.am	2008-04-04 04:02:30.000000000 -0500
+++ src/evolution-data-server-2.22.2/calendar/backends/caldav/Makefile.am	2008-06-01 21:27:04.640625000 -0500
@@ -27,4 +27,4 @@
 
 
 libecalbackendcaldav_la_LDFLAGS =		\
-	-module -avoid-version
+	-module -avoid-version $(NO_UNDEFINED)
--- origsrc/evolution-data-server-2.22.2/camel/camel-file-utils.h	2008-04-04 04:01:59.000000000 -0500
+++ src/evolution-data-server-2.22.2/camel/camel-file-utils.h	2008-06-01 21:27:04.656250000 -0500
@@ -31,6 +31,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <time.h>
+#include <sys/time.h>
 #include <fcntl.h>
 
 #ifndef O_BINARY
--- origsrc/evolution-data-server-2.22.2/camel/camel-net-utils.c	2008-04-04 04:01:59.000000000 -0500
+++ src/evolution-data-server-2.22.2/camel/camel-net-utils.c	2008-06-01 21:27:04.703125000 -0500
@@ -28,6 +28,8 @@
 #include <errno.h>
 #include <pthread.h>
 #include <stdio.h>
+#include <string.h>
+#include <netinet/in.h>
 
 #include <glib.h>
 #include <glib/gi18n-lib.h>
@@ -59,6 +61,9 @@
 #undef gai_strerror
 #define gai_strerror my_gai_strerror
 
+#endif
+
+#ifdef G_PLATFORM_WIN32
 /* gai_strerror() is implemented as an inline function in Microsoft's
  * SDK, but mingw lacks that. So implement here. The EAI_* errors can
  * be handled with the normal FormatMessage() API,
@@ -68,7 +73,11 @@
 static const char *
 gai_strerror (int error_code)
 {
+#ifdef G_WITH_CYGWIN
+	gchar *msg = g_strerror (error_code);
+#else
 	gchar *msg = g_win32_error_message (error_code);
+#endif
 	GQuark quark = g_quark_from_string (msg);
 	const gchar *retval = g_quark_to_string (quark);
 
--- origsrc/evolution-data-server-2.22.2/camel/camel-tcp-stream.h	2008-04-04 04:01:59.000000000 -0500
+++ src/evolution-data-server-2.22.2/camel/camel-tcp-stream.h	2008-06-01 21:27:04.750000000 -0500
@@ -39,6 +39,7 @@
 #include <unistd.h>
 
 #include <camel/camel-stream.h>
+#include <camel/camel-net-utils.h>
 
 #define CAMEL_TCP_STREAM_TYPE     (camel_tcp_stream_get_type ())
 #define CAMEL_TCP_STREAM(obj)     (CAMEL_CHECK_CAST((obj), CAMEL_TCP_STREAM_TYPE, CamelTcpStream))
--- origsrc/evolution-data-server-2.22.2/camel/providers/sendmail/Makefile.am	2008-04-04 04:01:43.000000000 -0500
+++ src/evolution-data-server-2.22.2/camel/providers/sendmail/Makefile.am	2008-06-01 21:27:04.781250000 -0500
@@ -21,7 +21,13 @@
 noinst_HEADERS =		\
 	camel-sendmail-transport.h
 
-libcamelsendmail_la_LDFLAGS = -avoid-version -module
+libcamelsendmail_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
+
+libcamelsendmail_la_LIBADD = \
+	$(top_builddir)/libedataserver/libedataserver-${API_VERSION}.la \
+	$(top_builddir)/camel/libcamel-provider-1.2.la          \
+	$(top_builddir)/camel/libcamel-1.2.la               \
+	$(CAMEL_LIBS)
 
 libcamelsendmail_la_LIBADD = \
 	$(top_builddir)/camel/libcamel-1.2.la				\
--- origsrc/evolution-data-server-2.22.2/camel/providers/smtp/camel-smtp-transport.c	2008-04-04 04:01:39.000000000 -0500
+++ src/evolution-data-server-2.22.2/camel/providers/smtp/camel-smtp-transport.c	2008-06-01 21:27:04.812500000 -0500
@@ -973,9 +973,11 @@
 	if (camel_getnameinfo (addr, addrlen, &name, NULL, NI_NUMERICHOST, NULL) != 0) {
 		name = g_strdup ("localhost.localdomain");
 	} else {
+#ifdef AF_INET6
 		if (addr->sa_family == AF_INET6)
 			numeric = "IPv6:";
 		else
+#endif
 			numeric = "";
 	}
 
--- origsrc/evolution-data-server-2.22.2/configure.in	2008-05-25 05:36:21.000000000 -0500
+++ src/evolution-data-server-2.22.2/configure.in	2008-06-01 21:27:04.828125000 -0500
@@ -175,30 +175,46 @@
 
 AC_MSG_CHECKING([for Win32])
 case "$host" in
+*-cygwin*)
+    platform_win32=yes
+    os_win32=no
+    NO_UNDEFINED='-no-undefined'
+    SOCKET_LIBS=''
+    DL_LIB=''
+    SOFTOKN3_LIB=''
+    LIBEXECDIR_IN_SERVER_FILE="$libexecdir"
+	SOPREFIX=cyg
+    ;;
 *-mingw*)
+    platform_win32=yes
     os_win32=yes
     NO_UNDEFINED='-no-undefined'
     SOCKET_LIBS='-lws2_32 -ldnsapi'
     DL_LIB=''
     SOFTOKN3_LIB=''
     LIBEXECDIR_IN_SERVER_FILE='../../../libexec'
+	SOPREFIX=lib
     # No, don't use Windows XP -only API
     # AC_CACHE_VAL(have_addrinfo, [have_addrinfo=yes])
     # AC_DEFINE(_WIN32_WINNT, 0x501, [To get getaddrinfo etc declarations])
     ;;
-*)  os_win32=no
+*)  platform_win32=yes
+    os_win32=no
     NO_UNDEFINED=''
     SOCKET_LIBS=''
     DL_LIB='-ldl'
     SOFTOKN3_LIB='-lsoftokn3'
     LIBEXECDIR_IN_SERVER_FILE="$libexecdir"
+	SOPREFIX=lib
     ;;
 esac
 AC_MSG_RESULT([$os_win32])
+AM_CONDITIONAL(PLATFORM_WIN32, [test $platform_win32 = yes])
 AM_CONDITIONAL(OS_WIN32, [test $os_win32 = yes])
 AC_SUBST(NO_UNDEFINED)
 AC_SUBST(SOCKET_LIBS)
 AC_SUBST(LIBEXECDIR_IN_SERVER_FILE)
+AC_SUBST(SOPREFIX)
 
 AC_CHECK_HEADERS(pthread.h semaphore.h sys/wait.h)
 AC_CHECK_FUNCS(fsync strptime strtok_r)
@@ -238,8 +254,8 @@
   dynamic_libdb=yes, dynamic_libdb=no)
 
 if test "x${dynamic_libdb}" = "xyes"; then
-        DB_CFLAGS="-I$withval/include"
-        DB_LIBS="-L$withval/lib -ldb"
+        DB_CFLAGS="-I$withval/include/db4.1"
+        DB_LIBS="-L$withval/lib -ldb-4.1"
 
         AC_MSG_CHECKING([Berkeley DB])
         save_cflags=$CFLAGS; CFLAGS=$DB_CFLAGS
@@ -1102,24 +1118,24 @@
 
 msg_krb5="no"
 if test "x${with_krb5}" != "xno"; then
-	LDFLAGS_save="$LDFLAGS"
+	LIBS_save="$LIBS"
 	
 	mitlibs="-lkrb5 -lk5crypto -lcom_err -lgssapi_krb5"
 	heimlibs="-lkrb5 -lcrypto -lasn1 -lcom_err -lroken -lgssapi"
 	sunlibs="-lkrb5 -lgss"
 	AC_CACHE_CHECK([for Kerberos 5], ac_cv_lib_kerberos5,
 	[
-		LDFLAGS="$LDFLAGS -L$with_krb5_libs $mitlibs"
+		LIBS="$LIBS -L$with_krb5_libs $mitlibs"
 		AC_TRY_LINK([#include <krb5.h>],krb5_init_context, ac_cv_lib_kerberos5="$mitlibs",
 		[
-			LDFLAGS="$LDFLAGS_save -L$with_krb5_libs $heimlibs"
+			LIBS="$LIBS_save -L$with_krb5_libs $heimlibs"
 			AC_TRY_LINK_FUNC(krb5_init_context, ac_cv_lib_kerberos5="$heimlibs", 
 			[
-				LDFLAGS="$LDFLAGS_save -L$with_krb5_libs $sunlibs"
+				LIBS="$LIBS_save -L$with_krb5_libs $sunlibs"
 				AC_TRY_LINK_FUNC(krb5_init_context, ac_cv_lib_kerberos5="$sunlibs", ac_cv_lib_kerberos5="no")
 			])
 		])
-		LDFLAGS="$LDFLAGS_save"
+		LIBS="$LIBS_save"
 	])
 	if test "$ac_cv_lib_kerberos5" != "no"; then
 		AC_DEFINE(HAVE_KRB5,1,[Define if you have Krb5])
@@ -1168,32 +1184,32 @@
 
 msg_krb4="no"
 if test "x${with_krb4}" != "xno"; then
-	LDFLAGS_save="$LDFLAGS"
+	LIBS_save="$LIBS"
 	AC_CACHE_CHECK(for Kerberos 4, ac_cv_lib_kerberos4,
 	[
 		ac_cv_lib_kerberos4="no"
 
 		mitcompatlibs="-lkrb4 -ldes425 -lkrb5 -lk5crypto -lcom_err"
 		# Look for MIT krb5 compat krb4
-		LDFLAGS="$LDFLAGS -L$with_krb4_libs $mitcompatlibs"
+		LIBS="$LIBS -L$with_krb4_libs $mitcompatlibs"
 		AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="$mitcompatlibs")
 		
 		if test "$ac_cv_lib_kerberos4" = "no"; then
 			# Look for KTH krb4
-			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs -lkrb -lcrypto -lcom_err -lroken"
+			LIBS="$LIBS_save -L$with_krb4_libs -lkrb -lcrypto -lcom_err -lroken"
 			AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb -lcrypto -lcom_err -lroken")
 		fi
 		if test "$ac_cv_lib_kerberos4" = "no"; then
 			# Look for old MIT krb4
-			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs -lkrb"
+			LIBS="$LIBS_save -L$with_krb4_libs -lkrb"
 			AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb",
 			[
-				LDFLAGS="$LDFLAGS -ldes"
+				LIBS="$LIBS -ldes"
 				AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb -ldes")
 			])
 		fi
 	])
-	LDFLAGS="$LDFLAGS_save"
+	LIBS="$LIBS_save"
 	if test "$ac_cv_lib_kerberos4" != "no"; then
 		AC_DEFINE(HAVE_KRB4,1,[Define if you have Krb4])
 		msg_krb4="yes"
@@ -1569,9 +1585,6 @@
 if test $enable_calendar = yes; then
         AC_CONFIG_SUBDIRS(calendar/libical)
 fi
-if test $dynamic_libdb = no; then
-        AC_CONFIG_SUBDIRS(libdb/dist)
-fi
 AC_OUTPUT([
 Makefile
 evolution-data-server-zip
--- origsrc/evolution-data-server-2.22.2/servers/exchange/storage/Makefile.am	2008-04-04 04:02:09.000000000 -0500
+++ src/evolution-data-server-2.22.2/servers/exchange/storage/Makefile.am	2008-06-01 21:27:04.859375000 -0500
@@ -1,6 +1,6 @@
 # Fix the code to not use E_DATA_SERVER_UI_CFLAGS
 
-if OS_WIN32
+if PLATFORM_WIN32
 WIN32_BOOTSTRAP_LIBS =					\
 	$(top_builddir)/win32/libedataserverui-1.2.la
 endif
@@ -20,6 +20,7 @@
 	$(E_DATA_SERVER_UI_CFLAGS)				\
 	-I$(top_builddir)/servers/exchange/lib			\
 	-I$(top_srcdir)						\
+	-I$(top_builddir)/servers/exchange/lib			\
 	-I$(top_srcdir)/servers/exchange/lib			\
 	-I$(top_srcdir)/servers/exchange/xntlm
 
--- origsrc/evolution-data-server-2.22.2/win32/Makefile.am	2008-04-04 04:02:26.000000000 -0500
+++ src/evolution-data-server-2.22.2/win32/Makefile.am	2008-06-01 21:27:04.921875000 -0500
@@ -2,7 +2,7 @@
 	dummy.la			\
 	libedataserverui.def
 
-if OS_WIN32
+if PLATFORM_WIN32
 BOOTSTRAP_LIBS =			\
 	libedataserverui-1.2.la
 endif
@@ -12,8 +12,8 @@
 libedataserverui_current_minus_age = `expr $(LIBEDATASERVERUI_CURRENT) - $(LIBEDATASERVERUI_AGE)`
 
 libedataserverui-1.2.la: dummy.la libedataserverui.def
-	sed -e s!%DLL%!libedataserverui-1.2-$(libedataserverui_current_minus_age)! -e s!%LIB%!libedataserverui-1.2! -e s!%PFX%!$(prefix)! <dummy.la >$@
+	sed -e s!%DLL%!$(SOPREFIX)edataserverui-1.2-$(libedataserverui_current_minus_age)! -e s!%LIB%!libedataserverui-1.2! -e s!%PFX%!$(prefix)! < $(srcdir)/dummy.la >$@
 	mkdir -p .libs
-	dlltool --output-lib=.libs/libedataserverui-1.2.dll.a --dllname=libedataserverui-1.2-$(libedataserverui_current_minus_age).dll --input-def=libedataserverui.def
+	dlltool --output-lib=.libs/libedataserverui-1.2.dll.a --dllname=$(SOPREFIX)edataserverui-1.2-$(libedataserverui_current_minus_age).dll --input-def=$(srcdir)/libedataserverui.def
 
 CLEANFILES = $(BOOTSTRAP_LIBS)
